src/esssans/normalization.py

```diff
@@ -64,6 +65,68 @@ def solid_angle_rectangular_approximation(
     )
 
 
+def solid_angle(
+    data: CalibratedMaskedData[RunType],
+    detector: DetectorGeometry[RunType],
+) -> SolidAngle[RunType]:
+    """
+    Solid angle for cylindrical pixels.
+
+    Note that the approximation is valid when the distance from sample
+    to pixel is much larger than the length or the radius of the pixels.
+
+    Parameters
+    ----------
+    data:
+        The DataArray that contains the positions of the detector pixels in coords.
+        The positions must be in the sample reference frame.
+
+    detector:
+        The DataGroup that contains the geometry of the detector.
+        Must contain pixel_shape and the transformation from the
+        local reference frame of the detector to the sample reference frame.
+
+    Returns
+    -------
+    :
+        The solid angle of the detector pixels, as viewed from the sample position.
+    """
+    face_1_center, face_1_edge, face_2_center = (
+        detector['pixel_shape']['vertices']['vertex', i] for i in range(3)
+    )
+    cylinder_axis = (
+        detector['transform'] * face_2_center - detector['transform'] * face_1_center
```

Comment:
Now I see why you called it `DetectorGeometry`. In practice this is the parent group of the raw data. I don't think we should expose this detail or any of the names (`pixel_shape` and `transform`) here. Those names can vary, depending on how files are created. This function should takes `pixel_shape` (or `cylinder_geometry`) and `transform` as separate inputs.